### PR TITLE
Supporting npm5.  Saving lock file.  Adjusted local npm tests.

### PR DESCRIPTION
**Pull request by Blackbaud-BobbyEarl**
This is combining the efforts of  https://github.com/blackbaud/skyux-builder/pull/260 and https://github.com/blackbaud/skyux-builder/pull/272

### CODE REVIEW

File: cli/utils/ts-linter.js [hunk 1]
```diff
@@ -19,6 +19,10 @@ function lintSync() {
   logger.info('Starting TSLint...');
 
   const spawnResult = spawn.sync('./node_modules/.bin/tslint', flags);
+  if (spawnResult.error) {
```
Comment (Blackbaud-SteveBrush) [inline cli/utils/ts-linter.js]: Thank you. I had to add this a few times (locally), but didn't get the opportunity to put it in a separate branch.

File: config/protractor/protractor-dev.conf.js [hunk 2]
```diff
@@ -9,6 +9,82 @@ const SpecReporter = require('jasmine-spec-reporter').SpecReporter;
...
+/**
+ * Copies current builder dependencies to cloned template.
+ */
+function mergeDependencies() {
+  const builderJson = fs.readJSONSync('package.json');
+  const templateFilename = path.resolve(common.tmp, 'package.json');
+  const templateJson = fs.readJSONSync(templateFilename);
+
+  for (let key in builderJson.dependencies) {
+    if (builderJson.dependencies.hasOwnProperty(key)) {
```
Comment (Blackbaud-SteveBrush) [inline config/protractor/protractor-dev.conf.js]: Non-blocking, but `Object.keys(` would let you skip this `if` check...
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/keys

```
Object.keys(builderJson.dependencies).forEach((key) => {
  templateJson.dependencies[key] = builderJson.dependencies[key];
});
```

File: config/protractor/protractor-dev.conf.js [hunk 3]
```diff
@@ -23,32 +99,27 @@ let config = {
...
-            url,
-            common.tmp
-          ]))
-          .then(() => common.exec(`npm`, [`i`, '--only=prod'], common.cwdOpts))
-          .then(() => common.exec(`npm`, [`i`, `../`], common.cwdOpts))
+          .then(cloneTemplate)
+          .then(mergeDependencies)
+          .then(copyLocks)
+          .then(() => common.exec(`npm`, [`install`, '--only=prod'], common.cwdOpts))
+          .then(copyBuilder)
```
Comment (Blackbaud-SteveBrush) [inline config/protractor/protractor-dev.conf.js]: This is SO much easier to piece-through.

File: config/protractor/protractor-dev.conf.js (hunk 2 above)
Comment (Blackbaud-BobbyEarl) [reply config/protractor/protractor-dev.conf.js]: Great suggestion.

File: cli/utils/ts-linter.js [hunk 4]
```diff
@@ -19,6 +19,10 @@ function lintSync() {
...
+    logger.error(spawnResult.error.message);
+    return process.exit(1);
```
Comment (Blackbaud-SteveBrush) [inline cli/utils/ts-linter.js]: I cloned [the template](https://github.com/blackbaud/skyux-template); and pointed builder to `blackbaud/skyux-builder#npm5-v2`.

After installing the dependencies, I then typed `skyux build -s` and got an error from the tslinter: 

```
error: spawnSync ./node_modules/.bin/tslint ENOENT
```

I think this error is saying that `.bin/tslint` doesn't exist in that directory (relative to the cwd).

I ran across this issue in my `npm5` branch, too, and couldn't figure out a good solution.

Comment (Blackbaud-SteveBrush) [reply cli/utils/ts-linter.js]: FYI, I'm running npm v4.6.1

**Comment by Blackbaud-BobbyEarl** [comment]
I have mangled this PR with a bad merge.  Luckily I was able to use `git reflog` to retrieve my work, but I'm still going to close this in favor of a clean PR.

**Pull request closed without merging**